win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1577,16 +1593,34 @@ IntGdiAddFontResourceEx(PUNICODE_STRING FileName, DWORD Characteristics,
     PVOID Buffer = NULL;
     IO_STATUS_BLOCK Iosb;
     PVOID SectionObject;
-    SIZE_T ViewSize = 0;
+    SIZE_T ViewSize = 0, Length;
     LARGE_INTEGER SectionSize;
     OBJECT_ATTRIBUTES ObjectAttributes;
     GDI_LOAD_FONT LoadFont;
     INT FontCount;
     HANDLE KeyHandle;
+    UNICODE_STRING PathName;
+    WCHAR szPath[MAX_PATH];
     static const UNICODE_STRING TrueTypePostfix = RTL_CONSTANT_STRING(L" (TrueType)");
 
+    /* Build PathName */
+    Length = wcslen(SharedUserData->NtSystemRoot);
+    FileName->Buffer[FileName->Length / sizeof(WCHAR)] = 0;
+    if (RtlCompareMemory(FileName->Buffer, SharedUserData->NtSystemRoot,
```

Comment:
What if the filename starts with systemroot in all lowercase? (there is _wcsnicmp to deal with case insensitivity, and where you specify also an explicit count of characters: for example here it would be `min(Length, FileName->Length/sizeof(WCHAR))` ).